packages/gatsby/src/query/index.js

```diff
@@ -68,48 +145,55 @@ const createStaticQueryJob = (state, queryId) => {
   }
 }
 
-const processStaticQueries = async (
-  queryIds,
-  { state, activity, graphqlRunner, graphqlTracing }
-) => {
-  state = state || store.getState()
-  await processQueries(
-    queryIds.map(id => createStaticQueryJob(state, id)),
-    {
-      activity,
-      graphqlRunner,
-      graphqlTracing,
-    }
-  )
+function onDevelopStaticQueryDone({ job, result }) {
+  websocketManager.emitStaticQueryData({
+    result,
+    id: job.hash,
+  })
 }
 
-const processPageQueries = async (
+async function processStaticQueries(
   queryIds,
   { state, activity, graphqlRunner, graphqlTracing }
-) => {
-  state = state || store.getState()
-  // Make sure we filter out pages that don't exist. An example is
-  // /dev-404-page/, whose SitePage node is created via
-  // `internal-data-bridge`, but the actual page object is only
-  // created during `gatsby develop`.
-
-  const jobs = []
-  queryIds.forEach(id => {
-    const page = state.pages.get(id)
-    if (page) {
-      const job = createPageQueryJob(state, page)
-      jobs.push(job)
-    }
+) {
+  return processQueries({
+    queryIds,
+    createJobFn: createStaticQueryJob,
+    onQueryDone:
+      process.env.NODE_ENV === `production`
+        ? undefined
+        : onDevelopStaticQueryDone,
+    state,
+    activity,
+    graphqlRunner,
+    graphqlTracing,
   })
+}
 
-  await processQueries(jobs, {
+async function processPageQueries(
+  queryIds,
+  { state, activity, graphqlRunner, graphqlTracing }
+) {
+  return processQueries({
+    queryIds,
+    createJobFn: createPageQueryJob,
+    state,
     activity,
     graphqlRunner,
     graphqlTracing,
   })
 }
 
-const createPageQueryJob = (state, page) => {
+function createPageQueryJob(state, queryId) {
+  const page = state.pages.get(queryId)
+
+  // Make sure we filter out pages that don't exist. An example is
+  // /dev-404-page/, whose SitePage node is created via
+  // `internal-data-bridge`, but the actual page object is only
+  // created during `gatsby develop`.
+  if (!page) {
+    return undefined
+  }
```

Comment:
similar question here? this doesn't seem related to `fastq` swap?

Reply:
Here is what we had before:

https://github.com/gatsbyjs/gatsby/blob/dad0032db4ea24f6923033906e3b29552f05c7ed/packages/gatsby/src/query/index.js#L91-L103

But now we've moved job creation to worker itself and so need to account for this in the createJob function itself.